### ISSUE TITLE
Status output of eagerly loaded repository rule not seen in Bazel 6 when running bazel build

### ISSUE THREAD

**Issue by gregmagolan** [bazelbuild/bazel#16978]
### Description of the bug:

The status of repository rules that are eagerly loaded in the WORKSPACE file are not seen
when running `bazel build`

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Clone https://github.com/gregmagolan/bazel_6_repo_terminal_output_repro and run `./repro.sh`

### Which operating system are you running Bazel on?

Observed on MacOS and Ubuntu

### What is the output of `bazel info release`?

release 6.0.0rc4

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
https://github.com/gregmagolan/bazel_6_repo_terminal_output_repro
```

### Have you found anything relevant by searching the web?

Nothing

### Any other information, logs, or outputs that you want to share?

This issue affects the DX of the rules_js `npm_translate_lock` repository rule since when this invalidates it can take a noticeable amount of time to execute in repositories with a large set of npm deps.

**Comment by meisterT** [comment]
I bisected this down to https://github.com/bazelbuild/bazel/commit/943e8cfb9970ca86f88ba60c944601573b366db5

**Comment by fmeum** [comment]
This seems to be a duplicate of https://github.com/bazelbuild/bazel/issues/16927 and (possibly) https://github.com/bazelbuild/bazel/issues/16338.

**Comment by meteorcloudy** [comment]
I'll close this as it's a duplicate of https://github.com/bazelbuild/bazel/issues/16927. Due to https://github.com/bazelbuild/bazel/issues/16927#issuecomment-1339638708, we probably will try to fix this in 6.1 instead of blocking the 6.0 release.
